sched/scheduler/task_scheduler.go

```diff
@@ -39,21 +39,24 @@ func getTaskAssignments(cs *clusterState, jobs []*jobState,
 	}
 	defer stat.Latency(stats.SchedTaskAssignmentsLatency_ms).Time().Stop()
 
-	if config == nil {
-		config = &SchedulerConfig{
-			SoftMaxSchedulableTasks: DefaultSoftMaxSchedulableTasks,
-		}
-	}
-
 	// Exit if there are no unscheduled tasks.
+	totalOutstandingTasks := 0
 	totalUnschedTasks := 0
 	for _, j := range jobs {
+		totalOutstandingTasks += (len(j.Tasks) - j.TasksCompleted)
 		totalUnschedTasks += (len(j.Tasks) - j.TasksCompleted - j.TasksRunning)
 	}
 	if totalUnschedTasks == 0 {
 		return nil, nil
 	}
 
+	// Udate SoftMaxSchedulableTasks based on number of healthy nodes and the total number of tasks.
+	// Setting the max to num healthy nodes means that each job can be fully scheduled.
+	if config == nil {
+		config = &SchedulerConfig{}
+	}
+	config.SoftMaxSchedulableTasks = max(config.SoftMaxSchedulableTasks, totalOutstandingTasks, len(cs.nodes))
```

Comment:
Still a little concerned about this line - seems weird that it never resets & is always increasing.

Reply:
Ok, I took that part out since it appears to be more confusing than useful.